packages/suite-native-atoms/src/Switch.tsx

```diff
@@ -1,55 +1,90 @@
-import React from 'react';
-import { TouchableOpacity } from 'react-native';
+import React, { useEffect } from 'react';
+import Animated, {
+    Easing,
+    useAnimatedStyle,
+    useSharedValue,
+    interpolateColor,
+    withTiming,
+} from 'react-native-reanimated';
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Box } from './Box';
+import { Pressable } from 'react-native';
 
 type SwitchProps = {
     isChecked: boolean;
     onChange: (value: boolean) => void;
     isDisabled?: boolean; // Functionality of disabled works but styles are not implemented yet (waiting for design)
 };
 
-type SwitchStyleProps = {
-    isChecked: boolean;
-};
-const wrapperStyle = prepareNativeStyle<SwitchStyleProps>((utils, { isChecked }) => ({
-    width: 44,
-    height: 24,
-    justifyContent: 'center',
-    alignItems: 'flex-start',
-    backgroundColor: utils.colors.gray400,
+const SWITCH_CONTAINER_WIDTH = 44;
+const SWITCH_CONTAINER_HEIGHT = 24;
+const SWITCH_CIRCLE_SIZE = 20;
+const SWITCH_CIRCLE_MARGIN = 2;
+const SWITCH_CIRCLE_TRACK_WIDTH =
+    SWITCH_CONTAINER_WIDTH - SWITCH_CIRCLE_SIZE - SWITCH_CIRCLE_MARGIN * 2;
+
+const switchContainerStyle = prepareNativeStyle(utils => ({
+    width: SWITCH_CONTAINER_WIDTH,
+    height: SWITCH_CONTAINER_HEIGHT,
     borderColor: utils.colors.forest,
     borderRadius: utils.borders.radii.round,
-    extend: {
-        condition: isChecked,
-        style: {
-            alignItems: 'flex-end',
-            backgroundColor: utils.colors.green,
-        },
-    },
+    flexDirection: 'row',
 }));
 
-const SWITCH_CIRCLE_SIZE = 20;
-
 const switchCircleStyle = prepareNativeStyle(utils => ({
     width: SWITCH_CIRCLE_SIZE,
     height: SWITCH_CIRCLE_SIZE,
     backgroundColor: utils.colors.white,
     borderRadius: utils.borders.radii.round,
-    margin: 2,
+    margin: SWITCH_CIRCLE_MARGIN,
+    alignSelf: 'center',
 }));
 
+const useAnimationStyles = ({ isChecked }: Pick<SwitchProps, 'isChecked'>) => {
+    const trackWidth = !isChecked ? 0 : SWITCH_CIRCLE_TRACK_WIDTH;
+    const { utils } = useNativeStyles();
+    const translateX = useSharedValue(trackWidth);
+
+    useEffect(() => {
+        translateX.value = withTiming(trackWidth, {
+            duration: 150,
+            easing: Easing.out(Easing.cubic),
+        });
+    }, [trackWidth, translateX]);
+
+    const animatedSwitchCircleStyle = useAnimatedStyle(() => ({
+        transform: [{ translateX: translateX.value }],
+    }));
+
+    const animatedSwitchContainerStyle = useAnimatedStyle(() => ({
+        backgroundColor: interpolateColor(
+            translateX.value,
+            [0, SWITCH_CIRCLE_TRACK_WIDTH],
+            [utils.colors.gray400, utils.colors.green],
+        ),
+    }));
+
+    return {
+        animatedSwitchCircleStyle,
+        animatedSwitchContainerStyle,
+    };
+};
```

Comment:
Just question for future animations - this is pretty small component but animation adds another 30-40 lines of codes? Maybe we could discuss separation of animation / styles. Or do you like to have it all in 1 file?

Reply:
I understand the functionality. What my question was is that this is basically just a hook for this component and doesn't need to be inside this file. So if we have some component that will have complicated animations as you mentioned, it could be overwhelming to keep it all in one file with styles, render method, types and possibly utils functions or lifecycle methods. It's fine with me, but this is basically a hook that we could basically use with another similar structured cmp if I'm not mistaken. So I was curious if we shouldn't move it elsewhere maybe?

Reply:
Approved the PR, so this is really just to kick off some debate about the code style with animations (consider this thread resolved, if you want to merge ASAP).

Reply:
I prefer to keep all things related to component in one file, in case it's used for just that one component. Separating things to files leads to unccessary noise in file system and losing context because usually animations depedends on styles of component that you have just few lines away if you have everything in same files.

It makes sense to separate only in case when implementaion of animation is very long or will need to reuse some animations, then I can imagine we will have something like `animations/useButtonPressAnimation.ts` etc.

Reply:
Great! Good to know for future. Thanks